src/librender/mesh.cpp

```diff
@@ -890,8 +890,19 @@ MTS_VARIANT void Mesh<Float, Spectrum>::parameters_changed(const std::vector<std
     }
 }
 
+MTS_VARIANT void Mesh<Float, Spectrum>::set_grad_suspended(bool state) {
+    if constexpr (ek::is_diff_array_v<Float>) {
```

Comment:
Here, and in other places: call parent class implementation of `set_grad_suspended` (e.g. `Shape` in this case). I suppose `Shape` should then disable the gradients of the BSDF, no?

Reply:
In the current implementation, `set_grad_suspended` is called on every plugin that sits in `params.hierarchy`, i.e. everything that the `traverse` mechanism went through, so in this case `set_grad_suspended` is called separately on the BSDF.